### PR TITLE
IGNITE-22166 CLI for disaster recovery: reset-lost-partitions

### PR DESCRIPTION
**Pull request by Phillippko**
https://issues.apache.org/jira/browse/IGNITE-22166

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/cli/src/main/java/org/apache/ignite/internal/cli/commands/Options.java [hunk 1]
```diff
@@ -298,11 +298,21 @@ public static final class Constants {
         public static final String RECOVERY_ZONE_NAMES_OPTION = "--zones";
 
         public static final String RECOVERY_ZONE_NAMES_OPTION_DESC = "Names specifying zones to get partition states from. "
-                + "Case-sensitive, all zones if not set";
+                + "Case-sensitive, without quotes, all zones if not set";
+
+        public static final String RECOVERY_ZONE_NAME_OPTION = "--zone";
+
+        public static final String RECOVERY_ZONE_NAME_OPTION_DESC = "Name of the zone to reset partitions of. "
+                + "Case-sensitive, without quotes";
```
Comment (sashapolo) [inline modules/cli/src/main/java/org/apache/ignite/internal/cli/commands/Options.java:306]: what do you mean by "without quotes"?

File: modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/commands/recovery/ItResetPartitionsTest.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    @Test
+    public void testResetPartitionZoneNotFound() {
+        String unknownZone = "unknown_zone";
+
+        execute(CLUSTER_URL_OPTION, NODE_URL,
+                RECOVERY_TABLE_NAME_OPTION, QUALIFIED_TABLE_NAME,
+                RECOVERY_ZONE_NAME_OPTION, unknownZone,
+                RECOVERY_PARTITION_IDS_OPTION, "1,2");
+
+        assertErrOutputContains("Distribution zone is not found [zoneName=" + unknownZone + "]");
```
Comment (sashapolo) [inline modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/commands/recovery/ItResetPartitionsTest.java:76]: should be `was not found`

File: modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/commands/recovery/ItResetPartitionsCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import java.util.List;
+import org.apache.ignite.internal.cli.commands.recovery.reset.ResetPartitionsCommand;
+import org.apache.ignite.internal.util.CollectionUtils;
+
+/** Test class for {@link ResetPartitionsCommand}. */
+public class ItResetPartitionsCommandTest extends ItResetPartitionsTest {
+
+    @Override
+    protected void execute(String... args) {
+        String[] fullArgs = CollectionUtils.concat(List.of("recovery", "reset-partitions"), List.of(args)).toArray(String[]::new);
```
Comment (sashapolo) [inline modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/commands/recovery/ItResetPartitionsCommandTest.java:29]: There's an `ArrayUtils#concat` method

File: modules/cli/src/main/java/org/apache/ignite/internal/cli/commands/Options.java (hunk 1 above)
Comment (Phillippko) [reply modules/cli/src/main/java/org/apache/ignite/internal/cli/commands/Options.java:306]: In SQL commands user has to encapsulate names in quotes if they are not in upper_case ("zone_name"), but in REST api (therefore CLI) we decided not to use quotes

**Review by sashapolo — APPROVED** [review]

**Pull request merged**